File: C1.UWP.FlexChart/VB/FlexChart101/Converters.vb
Imports C1.Chart
Imports System.Collections.Generic
Imports Windows.UI.Xaml.Data
Imports Windows.UI.Xaml
Imports System.Linq
Imports System

Public Class EnumConverter
    Implements IValueConverter
    Public Function Convert(value As Object, targetType As Type, parameter As Object, language As String) As Object Implements IValueConverter.Convert
        Return value.ToString()
    End Function

    Public Function ConvertBack(value As Object, targetType As Type, parameter As Object, language As String) As Object Implements IValueConverter.ConvertBack
        Dim str As String = value.ToString()
        Select Case parameter.ToString()
            Case "Palette"
                Return CType([Enum].Parse(GetType(Palette), str), Palette)
            Case "Stacking"
                Return CType([Enum].Parse(GetType(Stacking), str), Stacking)
            Case "PieLabelPosition"
                Return CType([Enum].Parse(GetType(PieLabelPosition), str), PieLabelPosition)
            Case "LabelPosition"
                Return CType([Enum].Parse(GetType(LabelPosition), str), LabelPosition)
            Case "ChartSelectionMode"
                Return CType([Enum].Parse(GetType(ChartSelectionMode), str), ChartSelectionMode)
            Case "Position"
                Return CType([Enum].Parse(GetType(Position), str), Position)
            Case "FunnelChartType"
                Return CType([Enum].Parse(GetType(FunnelChartType), str), FunnelChartType)
        End Select
        Return Nothing
    End Function
End Class

Public Class ChartTypeConverter
    Inherits DependencyObject
    Implements IValueConverter
    Public Property ChartTypes() As Dictionary(Of ChartType, String)
        Get
            Return CType(GetValue(ChartTypesProperty), Dictionary(Of ChartType, String))
        End Get
        Set
            SetValue(ChartTypesProperty, Value)
        End Set
    End Property

    Public Shared ReadOnly ChartTypesProperty As DependencyProperty = DependencyProperty.Register("ChartTypes", GetType(Dictionary(Of ChartType, String)), GetType(ChartTypeConverter), New PropertyMetadata(Nothing))

    Public Function Convert(value As Object, targetType As Type, parameter As Object, language As String) As Object Implements IValueConverter.Convert
        Dim key As ChartType = CType(value, ChartType)
        Dim index As Integer = 0
        For Each chartType As KeyValuePair(Of ChartType, String) In ChartTypes
            If chartType.Key.Equals(key) Then
                Return index
            End If
            index += 1
        Next

        Return index
    End Function

    Public Function ConvertBack(value As Object, targetType As Type, parameter As Object, language As String) As Object Implements IValueConverter.ConvertBack
        Dim index As Integer = CType(value, Integer)
        Return ChartTypes.ElementAt(index).Key
    End Function
End Class

Public Class VisibilityConverter
    Implements IValueConverter
    Public Function Convert(value As Object, targetType As Type, parameter As Object, language As String) As Object Implements IValueConverter.Convert
        Dim visibility As SeriesVisibility = CType([Enum].Parse(GetType(SeriesVisibility), value.ToString()), SeriesVisibility)
        Return visibility = SeriesVisibility.Visible
    End Function

    Public Function ConvertBack(value As Object, targetType As Type, parameter As Object, language As String) As Object Implements IValueConverter.ConvertBack
        Return If(CType(value, Boolean), SeriesVisibility.Visible, SeriesVisibility.Legend)
    End Function
End Class

Public Class FormatConverter
    Implements IValueConverter

    Public Function Convert(value As Object, targetType As Type, parameter As Object, language As String) As Object Implements IValueConverter.Convert
        Return value
    End Function

    Public Function ConvertBack(value As Object, targetType As Type, parameter As Object, language As String) As Object Implements IValueConverter.ConvertBack
        Return Double.Parse(value.ToString())
    End Function
End Class

File: C1.UWP.FlexChart/CS/FlexChart101/Converters.cs
using System;
using System.Linq;
using Windows.UI.Xaml;
using Windows.UI.Xaml.Data;
using System.Collections.Generic;
using C1.Chart;

namespace FlexChart101
{
    public class EnumConverter : IValueConverter
    {
        public object Convert(object value, Type targetType, object parameter, string language)
        {
            return value.ToString();
        }

        public object ConvertBack(object value, Type targetType, object parameter, string language)
        {
            string str = value.ToString();
            switch (parameter.ToString())
            {
                case "Palette":
                    return (Palette)Enum.Parse(typeof(Palette), str);
                case "Stacking":
                    return (Stacking)Enum.Parse(typeof(Stacking), str);
                case "PieLabelPosition":
                    return (PieLabelPosition)Enum.Parse(typeof(PieLabelPosition), str);
                case "LabelPosition":
                    return (LabelPosition)Enum.Parse(typeof(LabelPosition), str);
                case "ChartSelectionMode":
                    return (ChartSelectionMode)Enum.Parse(typeof(ChartSelectionMode), str);
                case "Position":
                    return (Position)Enum.Parse(typeof(Position), str);
                case "FunnelChartType":
                    return (FunnelChartType)Enum.Parse(typeof(FunnelChartType), str);
            }
            return null;
        }
    }

    public class ChartTypeConverter : DependencyObject, IValueConverter
    {
        public Dictionary<ChartType, string> ChartTypes
        {
            get { return (Dictionary<ChartType, string>)GetValue(ChartTypesProperty); }
            set { SetValue(ChartTypesProperty, value); }
        }

        public static readonly DependencyProperty ChartTypesProperty =
            DependencyProperty.Register("ChartTypes", typeof(Dictionary<ChartType, string>), typeof(ChartTypeConverter), 
            new PropertyMetadata(null));

        public object Convert(object value, Type targetType, object parameter, string language)
        {
            var key = (ChartType)value;
            int index = 0;
            foreach (var chartType in ChartTypes)
            {
                if (chartType.Key.Equals(key))
                    return index;
                index++;
            }

            return index;
        }

        public object ConvertBack(object value, Type targetType, object parameter, string language)
        {
            int index = (int)value;
            return ChartTypes.ElementAt(index).Key;
        }
    }

    public class VisibilityConverter : IValueConverter
    {
        public object Convert(object value, Type targetType, object parameter, string language)
        {
            var visibility = (SeriesVisibility)Enum.Parse(typeof(SeriesVisibility), value.ToString());
            return visibility == SeriesVisibility.Visible;
        }

        public object ConvertBack(object value, Type targetType, object parameter, string language)
        {
            return (bool)value ? SeriesVisibility.Visible : SeriesVisibility.Legend;
        }
    }

    public class FormatConverter : IValueConverter
    {
        public object Convert(object value, Type targetType, object parameter, string language)
        {
            return value;
        }

        public object ConvertBack(object value, Type targetType, object parameter, string language)
        {
            return double.Parse(value.ToString());
        }
    }
}
